ms.go

```diff
@@ -0,0 +1,140 @@
+package ms
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/coreos/go-oidc/v3/oidc"
+	"os"
+	"strings"
+)
+
+const rolesSeparator = "."
+const rolesKey = "roles"
+
+// Service defines basic parameters of a microservice for a client.
+type Service struct {
+	Host string
+	Port string
+}
+
+// BaseServiceServer contains a port on which the server listens for GRPC requests and provides JWT token verification.
+type BaseServiceServer struct {
+	Port       string
+	authIssuer string
+	verifier   *oidc.IDTokenVerifier
+}
+
+// Claims contains claims from a token and allows roles-checking for authorization.
+type Claims struct {
+	PreferredUsername string                         `json:"preferred_username"`
+	ResourceAccess    map[string]map[string][]string `json:"resource_access"`
```

Comment:
Claims are basically a JSON and it's its structure.
Actually user of the lib doesn't need the direct access to `ResourceAccess` and `RealmAccess`.
Instead, we can store just a set of roles (strings) that are defined in the same way as a parameter of `HasRole`.
Will it be a more suitable solution?

Reply:
Let's work under the principle of least privilege wherever possible, so wherever it is possible and makes sense to hide information from "user" of something, we'll do it.

What you wrote sounds reasonable.

BTW: is this aligned with how Keycloak's authorization works?

Reply:
It is aligned with how Keycloak's authorization works.
There is no specific claim in JWT token that is responsible for authorization in RFC. I will extract the code that is responsible for translation claims into roles in a separate function, so it will be easier to expand it for another provider.